MdeModulePkg/Core/Dxe/Mem/Page.c

```diff
@@ -949,6 +949,21 @@ CoreConvertPagesEx (
       Entry = NULL;
     }
 
+    // MU_CHANGE [BEGIN]
+    // The below call may allocate pages which, if we're freeing memory (implied by
+    // the new type being EfiConventionalMemory), could cause the memory we're currently
+    // freeing to be allocated before we're done freeing it if CoreFreeMemoryMapStack()
+    // is called after AddRange(). So, if we are freeing, let's free the memory map
+    // stack before adding memory we're converting to the free list.
+    if (NewType == EfiConventionalMemory) {

```

Comment:
Should we also check if ChangingType is true here?

Reply:
I think so, yes. Updated the behavior to only change the FreeMemoryMapStack() ordering if we are changing the type.